typescript/tests/MockPlayer.spec.ts

```diff
@@ -0,0 +1,46 @@
+// use this test if you are familiar with mocks.
+
+import { Inventory } from '../src/Inventory';
+import { Stats } from '../src/Stats';
+import { SimpleEnemy } from '../src/SimpleEnemy';
+import { Damage } from '../src/Damage';
+import { Player } from '../src/Player';
+
+jest.mock('../src/Inventory', () => {
+    return {
+        Inventory: jest.fn().mockImplementation(() => {
+            return {/* TODO */}
+        }),
+    };
+});
+
+jest.mock('../src/Stats', () => {
+    return {
+        Stats: jest.fn().mockImplementation(() => {
+            return {/* TODO */};
+        }),
+    };
+});
+
+jest.mock('../src/SimpleEnemy', () => {
+    return {
+        SimpleEnemy: jest.fn().mockImplementation(() => {
+            return {/* TODO */};
+        }),
+    };
+});
+
```

Comment:
haven't used Jest in this way. I have used
```
    hardware = {
      read: jest.fn((a) => 0),
      write: jest.fn((a, v) => { })
    };

```
for mocks, i.e. mocked each method. Was simpler. Maybe your code is the way it is intended

Reply:
I tried mocking the functions only. But I need mock objects of Inventory and Stats to create the OUT, i.e. the player.
Or how would you do that?

Reply:
I used this part of the documentation: https://jestjs.io/docs/es6-class-mocks